### PR TITLE
[SW-886] Chore/migrate tests

### PR DESCRIPTION
**Pull request by martinkaintas**
Sorry for the big PR but changes are dependent on each other
🤞 hopefully this will be the last big PR of the migration 🤞

### CODE REVIEW

File: src/store/getters.js [hunk 1]
```diff
@@ -44,18 +44,25 @@ export default {
     return accounts[activeIdx] || {}; // TODO: Return null
   },
   networks({ userNetworks }) {
+    if (!userNetworks) return {};
```
Comment (CedrikNikita) [inline src/store/getters.js:47]: That is incorrect.

File: src/store/index.js [hunk 2]
```diff
@@ -20,6 +20,11 @@ import { defaultNetwork, NODE_STATUS_CONNECTING } from '../popup/utils/constants
 
 export default new Vuex.Store({
   state: {
+    popupProps: {
```
Comment (CedrikNikita) [inline src/store/index.js:23]: @peronczyk will be against adding something in vuex.

File: tests/e2e/integration/aex2-popups.js [hunk 3]
```diff
@@ -55,7 +55,7 @@ describe('Tests cases for AEX-2 popups', () => {
     popups.forEach((popup) => {
       cy.openAex2Popup(popup, popup === 'sign' && SCHEMA.TX_TYPE.spend)
         .get('[data-cy=deny]')
-        .click()
+        .click({ force: true })
```
Comment (CedrikNikita) [inline tests/e2e/integration/aex2-popups.js:58]: Why `{ force: true }` is needed?

File: tests/e2e/integration/aex2-popups.js [hunk 4]
```diff
@@ -65,7 +65,7 @@ describe('Tests cases for AEX-2 popups', () => {
     popups.forEach((popup) => {
       cy.openAex2Popup(popup, popup === 'sign' && SCHEMA.TX_TYPE.spend)
         .get('[data-cy=accept]')
-        .click()
+        .click({ force: true })
```
Comment (CedrikNikita) [inline tests/e2e/integration/aex2-popups.js:68]: IDEM

File: jest.config.js [hunk 5]
```diff
@@ -16,14 +16,18 @@ module.exports = {
   moduleNameMapper: {
     '^.*\\.svg\\?vue-component$': '<rootDir>/config/jest/EmptySvg.vue',
     '^.*\\.svg\\?skip-optimize$': '<rootDir>/config/jest/EmptySvg.vue',
+    '\\.(css|scss)$': 'identity-obj-proxy',
+
```
Comment (CedrikNikita) [inline jest.config.js:20]:
```suggestion
```

File: src/popup/components/Modals/ConfirmRawSign.vue [hunk 6]
```diff
@@ -1,5 +1,6 @@
 <template>
   <Modal
+    show
```
Comment (CedrikNikita) [inline src/popup/components/Modals/ConfirmRawSign.vue:3]: How was it working before without `show`?

File: tests/e2e/integration/auth.js [hunk 7]
```diff
@@ -5,15 +5,13 @@ describe('Test cases for login functionality', () => {
       .should('be.visible')
 
       .shouldRedirect('/', '/account')
-      .shouldRedirect('/import-account', '/account')
-      .shouldRedirect('/more/about/terms', '/more/about/terms')
+      .shouldRedirect('#/more/about/terms', '/more/about/terms')
```
Comment (CedrikNikita) [inline tests/e2e/integration/auth.js:8]: Why have we returned to the `#` infront of the the route? Are the tests run on the web?

File: src/store/plugins/pendingTransactionHandler.js [hunk 8]
```diff
@@ -38,7 +38,7 @@ export default async (store) => {
     }
   };
 
-  await watchUntilTruthy(() => store.getters['sdkPlugin/sdk']);
+  await watchUntilTruthy(() => store.state.sdkPlugin.ready);
```
Comment (CedrikNikita) [inline src/store/plugins/pendingTransactionHandler.js:41]: Why ths change is necessary?

File: src/store/getters.js [hunk 9]
```diff
@@ -44,18 +44,25 @@ export default {
...
     return [
       NETWORK_MAINNET,
       NETWORK_TESTNET,
       ...userNetworks.map((network, index) => ({ index, ...network })),
     ].reduce((acc, n) => ({ ...acc, [n.name]: n }), {});
   },
-  activeNetwork({ current: { network } }, { networks }) {
+  activeNetwork(networkRef, { networks }) {
+    if (!networkRef) return;
```
Comment (CedrikNikita) [inline src/store/getters.js:55]: Why are these changes needed?

File: src/manifest.json [hunk 10]
```diff
@@ -2,6 +2,7 @@
   "name": "Superhero",
   "description": "Superhero Wallet",
   "manifest_version": 2,
+  "content_security_policy": "default-src 'self' 'unsafe-inline'; script-src 'self' 'sha256-BCCVwoZgDUk/cw+Ln2/t3z9bm8qy+sICaZP5BWQIyUg='; connect-src *; font-src * data:; img-src * data:; style-src-elem *",
```
Comment (CedrikNikita) [inline src/manifest.json:5]:
```
Refused to execute inline script because it violates the following 
Content Security Policy directive: "script-src 'self' 'sha256-BCCVwoZgDUk/cw+Ln2/t3z9bm8qy+sICaZP5BWQIyUg='". Either the 'unsafe-inline' keyword, a hash ('sha256-LDP+OjgDpfArTDlfhpqHPU5mcov+qbvbnr657nRMBZY='), or a nonce ('nonce-...') is required to enable inline execution
Uncaught EvalError: Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following 
Content Security Policy directive: "script-src 'self' 'sha256-BCCVwoZgDUk/cw+Ln2/t3z9bm8qy+sICaZP5BWQIyUg='".
```

File: src/popup/components/Modals/ConfirmRawSign.vue (hunk 6 above)
Comment (martinkaintas) [reply src/popup/components/Modals/ConfirmRawSign.vue:3]: `show` was introduced in `feature/migrate-to-vue3` when we removed the Vue migration build. In Vue3 when `transition` is used as root it wont trigger animations the way we were using it [[see](https://v3-migration.vuejs.org/breaking-changes/transition-as-root.html)]. 
`show` was set to true through `modals.ts/openModal()` but some modals can be accessed directly (without being opened from `openModal()`) so they need the `show` prop otherwise they don't show up

File: src/store/plugins/pendingTransactionHandler.js (hunk 8 above)
Comment (martinkaintas) [reply src/store/plugins/pendingTransactionHandler.js:41]: In this case `await watchUntilTruthy(() => store.getters['sdkPlugin/sdk'])` never gets resolved

File: tests/e2e/integration/auth.js (hunk 7 above)
Comment (martinkaintas) [reply tests/e2e/integration/auth.js:8]: `vue/router` in develop is initialized with `mode: IS_WEB ? 'history' : 'hash',`
after migrating to Vue3 it is initialized with `history: IS_WEB ? createWebHistory() : createWebHashHistory(),`
So if `IS_WEB` is not set we are using hash history. When running tests we are not specifying `IS_WEB` but if we do then the `#` is not required anymore

Comment (martinkaintas) [reply tests/e2e/integration/auth.js:8]: Should I add `PLATFORM=web` when running e2e tests and remove the `#`?

File: src/popup/router/index.ts [hunk 11]
```diff
@@ -69,7 +69,9 @@ router.beforeEach(async (to, from, next) => {
     }[POPUP_TYPE];
 
     if (name !== to.name) {
-      next({ name, params: await getPopupProps() as Dictionary });
+      const params = await getPopupProps() as Dictionary;
```
Comment (CedrikNikita) [inline src/popup/router/index.ts:72]: All this params related changes can be removed after router mode will be set to `history`.

File: src/manifest.json (hunk 10 above)
Comment (martinkaintas) [reply src/manifest.json:5]: Are you executing `npm run build:extension` or `npm run watch`? When building for development webpack is using `eval()` which is not allowed
`npm run build:extension` should work

File: tests/e2e/integration/aex2-popups.js (hunk 3 above)
Comment (martinkaintas) [reply tests/e2e/integration/aex2-popups.js:58]: Cypress is claiming that the buttons are hidden, but they are visible. By using `{force: true }` we are telling cypress not to check for this.

File: src/popup/router/index.ts [hunk 12]
```diff
@@ -69,7 +69,9 @@ router.beforeEach(async (to, from, next) => {
     }[POPUP_TYPE];
 
     if (name !== to.name) {
-      next({ name, params: await getPopupProps() as Dictionary });
```
Comment (CedrikNikita) [inline src/popup/router/index.ts:72]: After that the saved `popupProps` would not be removed from the state, and will be `v-bind` to every mounted component which is bad.
Previous approach stoped working becuase of that https://github.com/vuejs/router/blob/main/packages/router/CHANGELOG.md#414-2022-08-22
We have several options:
1) We can use one of the approaches the `vue-router` team suggesting.
2) We can write a composable that will load this info and use it for every modal that can be accesed in popup.
3) We can use `JSON.stringify` to `app` object and then `JSON.parse` it back in the popup.

Also it means that every place in the project where prop is not defined in the query with new vue-router it is not working anymore.

**Review by CedrikNikita — CHANGES_REQUESTED** [review]

**Review by CedrikNikita — COMMENTED** [review]
There are `app` inside `app` in this branch (and i guess the migration itself).
image
image

File: tests/e2e/integration/aex2-popups.js (hunk 3 above)
Comment (CedrikNikita) [reply tests/e2e/integration/aex2-popups.js:58]: I know what for the `force` option for. The question is what is covering/why is force needed.

Comment (martinkaintas) [reply tests/e2e/integration/aex2-popups.js:58]: This is related to the duplicate `#app` issue. After fixing that it now works without the `force` option

File: src/popup/components/Modals/ConfirmRawSign.vue [hunk 13]
```diff
@@ -57,12 +58,14 @@
 
 <script>
```
Comment (CedrikNikita) [inline src/popup/components/Modals/ConfirmRawSign.vue:59]:
```suggestion
<script lang="ts">
```

**Review by CedrikNikita — CHANGES_REQUESTED** [review]
Currently both `cancel` and `confirm` buttons doesnt trigger anything in the web iframe popups.

File: .github/workflows/main.yml [hunk 14]
```diff
@@ -16,6 +16,8 @@ jobs:
         with:
           path: ~/.npm
           key: ${{ runner.os }}-${{ hashFiles('**/package-lock.json') }}
+      - name: Upgrade NPM
+        run: npm install -g npm
```
Comment (peronczyk) [inline .github/workflows/main.yml:20]: Why the pipeline needs to update the NPM each time? Can you add more light to this?

File: src/composables/popupProps.ts [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+
+interface IPopupPropsState {
+  popupProps: Ref<IPopupProps | null>
+  // eslint-disable-next-line no-unused-vars
+  setPopupProps: (props: IPopupProps) => void
+}
+
+const popupProps = ref<IPopupProps | null>(null);
+
+export function usePopupProps(): IPopupPropsState {
```
Comment (peronczyk) [inline src/composables/popupProps.ts:14]: I suggest to add a comment that would describe what are the popup props.

File: src/composables/transactionTx.ts [hunk 16]
```diff
@@ -115,8 +115,8 @@ export function useTransactionTx({
   const direction = computed(() => innerTx.value?.function === TX_FUNCTIONS.claim
     ? TX_DIRECTION.received
     : getTxDirection.value(
-        outerTx.value?.payerId ? outerTx.value : innerTx.value,
-        externalAddress
+      outerTx.value?.payerId ? outerTx.value : innerTx.value,
+      externalAddress
```
Comment (peronczyk) [inline src/composables/transactionTx.ts:119]: In my opinion the previous indentation was more accurate.

File: .github/workflows/main.yml (hunk 14 above)
Comment (davidyuk) [reply .github/workflows/main.yml:20]: Can you update node version instead? node@14 gets installed with npm@6, but node@18 with npm@9 👍 
https://nodejs.org/ro/download/releases

File: src/popup/locales/en.json [hunk 17]
```diff
@@ -116,7 +116,7 @@
     },
     "payloadInfo": {
       "title": "Payload message",
-      "msg": "A payload is an optional public message composed by the sender and sent to the recipient along with a blockchain transaction.<br /><br />Payload can be utilized as additional transaction identifier, destination tag (required by some centralized exchanges when depositing funds), or just to provide any extra information in free-form text."
+      "msg": "A payload is an optional public message composed by the sender and sent to the recipient along with a blockchain transaction.{'<'}br /{'>'}{'<'}br /{'>'}Payload can be utilized as additional transaction identifier, destination tag (required by some centralized exchanges when depositing funds), or just to provide any extra information in free-form text."
```
Comment (peronczyk) [inline src/popup/locales/en.json:119]: Why the square brackets are wrapped now? This makes the code to look really bad. Maybe there is a different solution for this?

File: src/popup/pages/Popups/Connect.vue [hunk 18]
```diff
@@ -1,12 +1,13 @@
...
   <Modal
+    show
     full-screen
     class="connect"
     data-cy="popup-aex2"
   >
     <TransactionInfo
       :custom-title="$t('pages.connectConfirm.title')"
-      :sender="{ name: appName, address: app.host, url: app.url }"
+      :sender="{ name: appName, address: app?.host, url: app?.url }"
```
Comment (peronczyk) [inline src/popup/pages/Popups/Connect.vue:10]: I suggest to move the data passed to the prop to the computed property and set the return type to enforce the proper structure.

File: src/popup/pages/Popups/Connect.vue [hunk 19]
```diff
@@ -119,20 +114,21 @@ export default defineComponent({
...
+        ...popupProps.value?.app,
         ...permission.value,
       });
-      props.resolve();
+      popupProps.value?.resolve();
     }
 
     function cancel() {
-      props.reject(new Error('Rejected by user'));
+      popupProps.value?.reject(new Error('Rejected by user'));
```
Comment (peronczyk) [inline src/popup/pages/Popups/Connect.vue:124]: Please use `RejectedByUserError` instead.

File: src/popup/pages/Popups/Connect.vue [hunk 20]
```diff
@@ -119,20 +114,21 @@ export default defineComponent({
...
     }
 
     return {
       POPUP_CONNECT_ADDRESS_PERMISSION,
       POPUP_CONNECT_TRANSACTIONS_PERMISSION,
       appName,
+      app: popupProps.value?.app,
```
Comment (peronczyk) [inline src/popup/pages/Popups/Connect.vue:131]: Please avoid setting values in the return block.

File: src/popup/pages/Popups/MessageSign.vue [hunk 21]
```diff
@@ -1,29 +1,34 @@
...
     data-cy="popup-aex2"
   >
     <TransactionInfo
       :custom-title="$t('pages.popupMessageSign.title')"
-      :sender="{ name: app.name, address: app.host, url: app.url }"
+      :sender="{
+        name: popupProps?.app?.name,
+        address: popupProps?.app?.host,
+        url: popupProps?.app?.url
+      }"
```
Comment (peronczyk) [inline src/popup/pages/Popups/MessageSign.vue:14]: I suggest to move this into composable and set the correct type for the return value to enforce correct data structure.

File: src/popup/pages/Popups/MessageSign.vue [hunk 22]
```diff
@@ -90,12 +89,13 @@ export default defineComponent({
     }));
 
     function cancel() {
-      props.reject(new Error('Rejected by user'));
+      popupProps.value?.reject(new Error('Rejected by user'));
```
Comment (peronczyk) [inline src/popup/pages/Popups/MessageSign.vue:92]: Please use `RejectedByUserError` instead. Probably you didn't rebased your branch for a while - on current develop this was already changed.

File: src/popup/router/index.ts [hunk 23]
```diff
@@ -69,7 +70,9 @@ router.beforeEach(async (to, from, next) => {
...
+      const params = await getPopupProps() as Dictionary;
```
Comment (peronczyk) [inline src/popup/router/index.ts:73]:
```suggestion
      const popupProps = await getPopupProps() as Dictionary;
```

File: src/store/plugins/pendingTransactionHandler.js (hunk 8 above)
Comment (peronczyk) [reply src/store/plugins/pendingTransactionHandler.js:41]: Instead I suggest to use the composable `const { getSdk } = useSdk()` and to ensure the sdk is accessible run `await getSdk()`.

File: src/types/index.ts [hunk 24]
```diff
@@ -645,8 +645,8 @@ export interface IKeyPair {
 }
 
 export interface ILabel {
-  text: string | LocaleMessages | TranslateResult,
-  customPending?: string | LocaleMessages | TranslateResult
+  text: string | TranslateResult,
+  customPending?: string | TranslateResult
```
Comment (peronczyk) [inline src/types/index.ts:649]: Please use semicolons at the end of the lines.

File: tests/e2e/support/commands.js [hunk 25]
```diff
@@ -11,7 +11,7 @@ import {
 } from '../../../src/popup/utils';
 
 Cypress.Commands.add('openPopup', (onBeforeLoad, route) => {
-  cy.visit(`${route ? `#${route}` : ''}`, { onBeforeLoad });
+  cy.visit(`${route || ''}`, { onBeforeLoad });
```
Comment (peronczyk) [inline tests/e2e/support/commands.js:14]:
```suggestion
  cy.visit(route || '', { onBeforeLoad });
```

File: tests/unit/snapshot.spec.js [hunk 26]
```diff
@@ -1,52 +1,73 @@
...
 import Index from '../../src/popup/pages/Index.vue';
 import About from '../../src/popup/pages/About.vue';
 import TermsOfService from '../../src/popup/pages/TermsOfService.vue';
 import PrivacyPolicy from '../../src/popup/pages/PrivacyPolicy.vue';
+import * as environment from '../../src/lib/environment';
 
-Object.assign(Vue.prototype, {
-  $t: () => 'locale-specific-text',
+const OLD_ENV = process.env;
+beforeEach(() => {
```
Comment (peronczyk) [inline tests/unit/snapshot.spec.js:9]: Please put empty line before this one :)

**Review by peronczyk — CHANGES_REQUESTED** [review]

File: src/store/plugins/pendingTransactionHandler.js (hunk 8 above)
Comment (martinkaintas) [reply src/store/plugins/pendingTransactionHandler.js:41]: This is also not working 🥲. It never gets resolved

File: src/popup/locales/en.json (hunk 17 above)
Comment (martinkaintas) [reply src/popup/locales/en.json:119]: when using `@intlify/vue-i18n-loader` html is not allowed in translation texts. When trying to build the extension we will get an error

File: src/store/plugins/pendingTransactionHandler.js (hunk 8 above)
Comment (peronczyk) [reply src/store/plugins/pendingTransactionHandler.js:41]: Any errors when doing `getSdk()`? It works perfectly in every other place.

Comment (martinkaintas) [reply src/store/plugins/pendingTransactionHandler.js:41]: No errors, the promise just doesn't resolve

**Review by peronczyk — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 14 above)
Comment (martinkaintas) [reply .github/workflows/main.yml]: This has been fixed. Some extra packages are required to be installed in order to use node v18 (v16 as well)

File: src/popup/components/Modals/ConfirmTransactionSign.vue [hunk 27]
```diff
@@ -308,23 +305,25 @@ export default defineComponent({
     }
 
     function cancel() {
-      props.reject(new Error('Rejected by user'));
+      popupProps.value?.reject(new Error('Rejected by user'));
```
[the remark this answers is not in the document]
Comment (martinkaintas) [reply src/popup/components/Modals/ConfirmTransactionSign.vue:311]: Errors related to rebase will be fixed in a separate PR. This error type does not exist in this branch

File: src/popup/components/Modals/ConfirmRawSign.vue [hunk 28]
```diff
@@ -73,35 +79,47 @@ export default {
...
+  setup() {
+    const { popupProps } = usePopupProps();
+    const { t } = useI18n();
+    const store = useStore();
+
+    const dataAsString = computed(() => {
+      if (typeof popupProps.value?.data === 'string') return popupProps.value?.data;
+      return Buffer.from(popupProps.value?.data as any).toString('hex');
+    });
+    const sender: ComputedRef<IAccountOverView> = computed(() => ({
```
[the remark this answers is not in the document]
Comment (martinkaintas) [reply src/popup/components/Modals/ConfirmRawSign.vue:91]: With all the refactoring I've forgotten how to write code 🫠

File: src/store/plugins/pendingTransactionHandler.js (hunk 8 above)
Comment (peronczyk) [reply src/store/plugins/pendingTransactionHandler.js:41]: OK, for now let's go with your approach. Soon we will remove Vuex.

**Review by peronczyk — APPROVED** [review]

File: .github/workflows/main.yml (hunk 14 above)
Comment (CedrikNikita) [reply .github/workflows/main.yml]: _You have jsdom (jest) dependent on an old version of canvas that tries to build under nodejs@18. Current fix is a workaround, it's better to update dependencies._
originally mentioned by @davidyuk

Comment (martinkaintas) [reply .github/workflows/main.yml]: `@vue/cli-plugin-unit-jest` doesn't support jest@28/29.  [see](https://github.com/vuejs/vue-cli/issues/7129)

Comment (davidyuk) [reply .github/workflows/main.yml]: Can you update jsdom only? The `canvas` package is only needed to emulate it for tests

Comment (martinkaintas) [reply .github/workflows/main.yml]: Updating `canvas` fixes the issue

Comment (davidyuk) [reply .github/workflows/main.yml]: Maybe `canvas` is not built for the env you have in gh actions 🤷‍♀️

Comment (martinkaintas) [reply .github/workflows/main.yml]: Turns out `npm ci` doesn't run locally either. So the issue still remains.
Yes, canvas is trying to build it self because it doesn't a prebuilt version, but then it fails. The packages I had added solved this.
This is also an official fix [[see](https://github.com/Automattic/node-canvas#compiling)]
I think adding the packages is the best option here

**Review by CedrikNikita — APPROVED** [review]
